packages/next/src/plugins/plugin.ts

```diff
@@ -0,0 +1,229 @@
+  projectRoot: string,
+  nextConfig: NextConfig,
+  options: NextPluginOptions
+) {
+  const targetDefaults = readTargetDefaultsForTarget(
+    options.buildTargetName,
+    context.nxJsonConfiguration.targetDefaults,
+    'nx:run-commands'
+  );
+
+  // Set output path here so that `withNx` can pick it up.
+  const targetConfig: TargetConfiguration = {
+    command: `next build`,
+    options: {
+      cwd: projectRoot,
+    },
+  };
+
+  if (targetDefaults?.outputs === undefined) {
+    targetConfig.outputs = [await getOutputs(projectRoot, nextConfig)];
+  }
+
+  if (targetDefaults?.cache === undefined) {
+    targetConfig.cache = true;
+  }
+
+  if (targetDefaults?.inputs === undefined) {
+    targetConfig.inputs = getInputs(namedInputs);
+  }
+
+  return targetConfig;
+}
+
+function getDevTargetConfig(projectRoot: string) {
+  const targetConfig: TargetConfiguration = {
+    command: `next dev`,
+    options: {
+      cwd: projectRoot,
+    },
+  };
+
+  return targetConfig;
+}
+
+function getStartTargetConfig(options: NextPluginOptions) {
+  const targetConfig: TargetConfiguration = {
+    command: `next start`,
+    options: {
+      cwd: `dist/{projectRoot}`,
+    },
+    dependsOn: [options.buildTargetName],
+  };
+
+  return targetConfig;
+}
+
+async function getOutputs(projectRoot, nextConfig) {
+  let dir = '.next';
+
+  if (typeof nextConfig === 'function') {
+    // Works for both async and sync functions.
+    const configResult = await Promise.resolve(
+      nextConfig(PHASE_PRODUCTION_BUILD, { defaultConfig: {} })
+    );
+    if (configResult?.distDir) {
+      dir = configResult?.distDir;
+    }
+  } else if (typeof nextConfig === 'object' && nextConfig?.distDir) {
+    // If nextConfig is an object, directly use its 'distDir' property.
+    dir = nextConfig.distDir;
+  }
+  return `{workspaceRoot}/dist/${projectRoot}/${dir}`;
+}
+
+function getNextConfig(
+  configFilePath: string,
+  context: CreateNodesContext
+): Promise<any> {
+  const resolvedPath = join(context.workspaceRoot, configFilePath);
+
+  /**
+   * Transpiled code for this is incorrect as require does not work on ESM modules.
+   * The code -> const module = await import(resolvedPath);
+   * Results in const module = await Promise.resolve(`${resolvedPath}`).then(s => require(s));
+   *
+   * ************ WorkAround ***************
+   * const module = await (Function(`return import("${resolvedPath}")`)() as Promise<any>);
+   * */
+
+  // const module = await (Function(`return import("${resolvedPath}")`)() as Promise<any>);
+  const module = require(resolvedPath);
+  return module.default ?? module;
+}
+
+function normalizeOptions(options: NextPluginOptions): NextPluginOptions {
+  options ??= {};
+  options.buildTargetName ??= 'build';
+  options.devTargetName ??= 'dev';
+  options.startTargetName ??= 'start';
+  return options;
+}
+
+function getInputs(
+  namedInputs: NxJsonConfiguration['namedInputs']
+): TargetConfiguration['inputs'] {
+  return [
```

Comment:
We realized that this will prevent `externalDependencies` from being added to inputs for build, since Nx generates with a default `build` entry in `nx.json`.

See Katerina's fix here https://github.com/nrwl/nx/pull/20528/files#diff-e21b84bc7a86a657a55ca6be0f7f8dbe1c2a3ff3475357e0de168fc775c987bbR174-R177